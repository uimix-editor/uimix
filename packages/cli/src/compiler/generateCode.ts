import { ProjectJSON } from "@uimix/node-data";
import { Project } from "@uimix/editor/src/models/Project";
import { loadProjectJSON } from "@uimix/editor/src/models/ProjectJSON";
import { formatTypeScript } from "../format.js";
import { CSSGenerator } from "./CSSGenerator.js";
import * as Y from "yjs";
import { ReactGenerator } from "./ReactGenerator.js";
import { dataUriToBuffer } from "data-uri-to-buffer";
import * as mime from "mime-types";

export async function generateCode(
  rootPath: string,
  projectJSON: ProjectJSON
): Promise<
  {
    filePath: string;
    content: string | Buffer;
  }[]
> {
  const ydoc = new Y.Doc();
  const project = new Project(ydoc);
  loadProjectJSON(ydoc, projectJSON);

  const imagesPath = ".uimix/images";

  const results: {
    filePath: string;
    content: string | Buffer;
  }[] = [];

  for (const [hash, image] of Object.entries(projectJSON.images ?? {})) {
    const decoded = dataUriToBuffer(image.url);
    const suffix = mime.extension(decoded.type) || "bin";
    results.push({
      filePath: `${imagesPath}/${hash}.${suffix}`,
      content: decoded,
    });
  }

  for (const page of project.pages.all) {
    const tsContent = formatTypeScript(
      new ReactGenerator({ rootPath, page, imagesPath }).render().join("\n")
    );
    const cssContent = new CSSGenerator(page).generate();

    results.push(
      {
        filePath: page.filePath + ".uimix.tsx",
        content: tsContent,
      },
      {
        filePath: page.filePath + ".uimix.css",
        content: cssContent,
      }
    );
  }

  return results;
}
