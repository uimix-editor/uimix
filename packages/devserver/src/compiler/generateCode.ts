import { ProjectJSON } from "@uimix/node-data";
import { Project } from "@uimix/editor/src/models/Project";
import { formatTypeScript } from "../format.js";
import { CSSGenerator } from "./CSSGenerator.js";
import * as Y from "yjs";
import { ReactGenerator } from "./ReactGenerator.js";

export function generateCode(
  projectJSON: ProjectJSON,
  imageFiles: string[]
): {
  "index.tsx": string;
  "index.css": string;
} {
  const ydoc = new Y.Doc();
  const project = new Project(ydoc);
  project.loadJSON(projectJSON);

  const tsContent = formatTypeScript(
    new ReactGenerator(project, imageFiles).render().join("\n")
  );
  const cssContent = new CSSGenerator(project).generate();

  return {
    "index.tsx": tsContent,
    "index.css": cssContent,
  };
}
