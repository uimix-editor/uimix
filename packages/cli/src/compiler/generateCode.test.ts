import { describe, expect, it } from "vitest";
import * as path from "path";
import * as fs from "fs";
import { generateCode } from "./generateCode";

describe(generateCode.name, () => {
  it("generates code", () => {
    const uimixPath = "../../../sandbox/src/uimix";
    const json = JSON.parse(
      fs.readFileSync(path.resolve(__dirname, uimixPath, "data.json"), "utf8")
    );
    const imageFiles = fs.readdirSync(
      path.resolve(__dirname, uimixPath, "images")
    );
    const code = generateCode(json, imageFiles);
    expect(code).toMatchSnapshot();
  });
});
