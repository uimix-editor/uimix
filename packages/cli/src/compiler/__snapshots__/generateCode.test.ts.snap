// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`generateCode > generates code 1`] = `
{
  "src/components.uimix.css": ".uimix-1Bvi2z3UPfi6W3L8nDt5VA {
box-sizing: border-box;
-webkit-font-smoothing: antialiased;
  position: relative;
  width: max-content;
  height: max-content;
  opacity: 1;
  overflow: visible;
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  justify-content: flex-start;
  row-gap: 0px;
  column-gap: 0px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  background: #E5E7EB;
  border-style: solid;
  border-color: transparent;
  border-top-width: 0px;
  border-right-width: 0px;
  border-bottom-width: 0px;
  border-left-width: 0px;
  border-top-left-radius: 0px;
  border-top-right-radius: 0px;
  border-bottom-right-radius: 0px;
  border-bottom-left-radius: 0px;
  box-shadow: none;
}
.uimix-1Bvi2z3UPfi6W3L8nDt5VA:hover {
  background: #3b82f6;
}
@media (max-width: 767px) {
.uimix-1Bvi2z3UPfi6W3L8nDt5VA {
  padding-left: 4px;
  padding-right: 4px;
  padding-top: 2px;
  padding-bottom: 2px;
}
}
.uimix-1Bvi2z3TfkSpYXW29daTN8 {
box-sizing: border-box;
-webkit-font-smoothing: antialiased;
  position: relative;
  width: max-content;
  height: max-content;
  opacity: 1;
  overflow: visible;
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  justify-content: flex-start;
  row-gap: 10px;
  column-gap: 10px;
  padding-left: 40px;
  padding-right: 20px;
  padding-top: 10px;
  padding-bottom: 30px;
  background: #FFFFFF;
  border-style: solid;
  border-color: transparent;
  border-top-width: 0px;
  border-right-width: 0px;
  border-bottom-width: 0px;
  border-left-width: 0px;
  border-top-left-radius: 0px;
  border-top-right-radius: 0px;
  border-bottom-right-radius: 0px;
  border-bottom-left-radius: 0px;
  box-shadow: none;
}
.uimix-1Bvi319YVxZELz4tEskN1S {
box-sizing: border-box;
-webkit-font-smoothing: antialiased;
  position: relative;
  width: 900px;
  height: 456px;
  opacity: 1;
  overflow: visible;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  row-gap: 0px;
  column-gap: 0px;
  padding-left: 0px;
  padding-right: 0px;
  padding-top: 0px;
  padding-bottom: 0px;
  background: #FFFFFF;
  border-style: solid;
  border-color: transparent;
  border-top-width: 0px;
  border-right-width: 0px;
  border-bottom-width: 0px;
  border-left-width: 0px;
  border-top-left-radius: 0px;
  border-top-right-radius: 0px;
  border-bottom-right-radius: 0px;
  border-bottom-left-radius: 0px;
  box-shadow: none;
}
.uimix-1BwWRgjHrhVbbuTV7m36kM {
box-sizing: border-box;
-webkit-font-smoothing: antialiased;
  position: relative;
  width: 400px;
  height: max-content;
  opacity: 1;
  overflow: visible;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-start;
  row-gap: 20px;
  column-gap: 20px;
  padding-left: 19px;
  padding-right: 19px;
  padding-top: 19px;
  padding-bottom: 19px;
  background: #ffffff;
  border-style: solid;
  border-color: transparent;
  border-top-width: 0px;
  border-right-width: 0px;
  border-bottom-width: 0px;
  border-left-width: 0px;
  border-top-left-radius: 20px;
  border-top-right-radius: 20px;
  border-bottom-right-radius: 20px;
  border-bottom-left-radius: 20px;
  box-shadow: none;
}
.uimix-1Bvi2z3UGTM64JtRT7Qc1S {
box-sizing: border-box;
-webkit-font-smoothing: antialiased;
  position: relative;
  margin-top: 0px;
  margin-right: 0px;
  margin-bottom: 0px;
  margin-left: 0px;
  width: max-content;
  height: max-content;
  opacity: 1;
  overflow: visible;
  white-space: break-spaces;
  display: flex;
  flex-direction: column;
  color: #111827;
  font-family: Inter;
  font-size: 16px;
  font-weight: 400;
  line-height: normal;
  letter-spacing: 0px;
  text-align: start;
  justify-content: flex-start;
}
.uimix-1Bvi2z3UPfi6W3L8nDt5VA:hover .uimix-1Bvi2z3UGTM64JtRT7Qc1S {
  color: #FFFFFF;
}
@media (max-width: 767px) {
.uimix-1Bvi2z3UGTM64JtRT7Qc1S {
  font-size: 14px;
}
}
.uimix-1Bvi2z3TpKXtTJLTRPTEd8 {
box-sizing: border-box;
-webkit-font-smoothing: antialiased;
  position: relative;
  margin-top: 0px;
  margin-right: 0px;
  margin-bottom: 0px;
  margin-left: 0px;
  flex: 1;
  min-width: 25px;
  max-width: 100px;
  height: calc(100% - 0px);
  min-height: 20px;
  max-height: 60px;
  opacity: 1;
  overflow: visible;
  display: block;
  padding-left: 0px;
  padding-right: 0px;
  padding-top: 0px;
  padding-bottom: 0px;
  background: #EF4444;
  border-style: solid;
  border-color: transparent;
  border-top-width: 0px;
  border-right-width: 0px;
  border-bottom-width: 0px;
  border-left-width: 0px;
  border-top-left-radius: 0px;
  border-top-right-radius: 0px;
  border-bottom-right-radius: 0px;
  border-bottom-left-radius: 0px;
  box-shadow: none;
}
.uimix-1Bvi2z3To8ddXo6R6ZftVD {
box-sizing: border-box;
-webkit-font-smoothing: antialiased;
  position: relative;
  margin-top: 0px;
  margin-right: 0px;
  margin-bottom: 0px;
  margin-left: 0px;
  width: 40px;
  height: 80px;
  opacity: 1;
  overflow: visible;
  display: block;
  padding-left: 0px;
  padding-right: 0px;
  padding-top: 0px;
  padding-bottom: 0px;
  background: #22C55E;
  border-style: solid;
  border-color: transparent;
  border-top-width: 0px;
  border-right-width: 0px;
  border-bottom-width: 0px;
  border-left-width: 0px;
  border-top-left-radius: 0px;
  border-top-right-radius: 0px;
  border-bottom-right-radius: 0px;
  border-bottom-left-radius: 0px;
  box-shadow: none;
}
.uimix-1BwspVKcGYvxVctYVzjPu6 {
box-sizing: border-box;
-webkit-font-smoothing: antialiased;
  position: relative;
  margin-top: 0px;
  margin-right: 0px;
  margin-bottom: 0px;
  margin-left: 0px;
  width: 40px;
  height: 80px;
  opacity: 1;
  overflow: visible;
  display: none;
  padding-left: 0px;
  padding-right: 0px;
  padding-top: 0px;
  padding-bottom: 0px;
  background: #22C55E;
  border-style: solid;
  border-color: transparent;
  border-top-width: 0px;
  border-right-width: 0px;
  border-bottom-width: 0px;
  border-left-width: 0px;
  border-top-left-radius: 0px;
  border-top-right-radius: 0px;
  border-bottom-right-radius: 0px;
  border-bottom-left-radius: 0px;
  box-shadow: none;
}
.uimix-1Bvi2z3UB1xUVg4Xz9WTpJ {
box-sizing: border-box;
-webkit-font-smoothing: antialiased;
  position: relative;
  margin-top: 0px;
  margin-right: 0px;
  margin-bottom: 0px;
  margin-left: 0px;
  width: 80px;
  height: 40px;
  opacity: 1;
  overflow: visible;
  display: block;
  padding-left: 0px;
  padding-right: 0px;
  padding-top: 0px;
  padding-bottom: 0px;
  background: #3b82f6;
  border-style: solid;
  border-color: transparent;
  border-top-width: 0px;
  border-right-width: 0px;
  border-bottom-width: 0px;
  border-left-width: 0px;
  border-top-left-radius: 0px;
  border-top-right-radius: 0px;
  border-bottom-right-radius: 0px;
  border-bottom-left-radius: 0px;
  box-shadow: none;
}
.uimix-1Bvi2z3U3SiK6r7gf9vh7N {
box-sizing: border-box;
-webkit-font-smoothing: antialiased;
  position: relative;
  margin-top: 10px;
  margin-right: 20px;
  margin-bottom: 30px;
  margin-left: 40px;
  width: max-content;
  height: max-content;
  opacity: 1;
  overflow: visible;
  white-space: break-spaces;
  display: flex;
  flex-direction: column;
  color: #000000;
  font-family: Inter;
  font-size: 20px;
  font-weight: 400;
  line-height: normal;
  letter-spacing: 0px;
  text-align: start;
  justify-content: flex-start;
}
.uimix-1BxSDQc9suV4UZUVEQsy6r {
  margin-top: 0px;
  margin-right: 0px;
  margin-bottom: 0px;
  margin-left: 0px;
}
.uimix-1BxSDQc9suV4UZUVEQsy6r-1Bvi2z3UGTM64JtRT7Qc1S {
}
.uimix-1Bvi31VKu8gfy6XzJ8xrHA {
box-sizing: border-box;
-webkit-font-smoothing: antialiased;
  position: relative;
  margin-top: 0px;
  margin-right: 0px;
  margin-bottom: 0px;
  margin-left: 0px;
  width: calc(100% - 0px);
  min-width: undefined;
  max-width: undefined;
  height: max-content;
  opacity: 1;
  overflow: visible;
}
.uimix-1Bvi365mtWZmXKDPeJTB4s {
box-sizing: border-box;
-webkit-font-smoothing: antialiased;
  position: relative;
  margin-top: 0px;
  margin-right: 0px;
  margin-bottom: 0px;
  margin-left: 0px;
  width: calc(100% - 0px);
  min-width: undefined;
  max-width: undefined;
  flex: 1;
  min-height: undefined;
  max-height: undefined;
  opacity: 1;
  overflow: visible;
  display: block;
  padding-left: 0px;
  padding-right: 0px;
  padding-top: 0px;
  padding-bottom: 0px;
  background: #FFFFFF;
  border-style: solid;
  border-color: transparent;
  border-top-width: 0px;
  border-right-width: 0px;
  border-bottom-width: 0px;
  border-left-width: 0px;
  border-top-left-radius: 0px;
  border-top-right-radius: 0px;
  border-bottom-right-radius: 0px;
  border-bottom-left-radius: 0px;
  box-shadow: none;
}
.uimix-1Bvi32d9RMi3ES5kk2UvuM {
box-sizing: border-box;
-webkit-font-smoothing: antialiased;
  position: absolute;
  left: 33px;
  top: 28px;
  width: max-content;
  height: max-content;
  opacity: 1;
  overflow: visible;
}
.uimix-1Bvi2z3UrU9VkawwAteXs8 {
  position: absolute;
  left: 160px;
  top: 28px;
  background: #BABABA;
}
.uimix-1Bvi2z3UrU9VkawwAteXs8-1Bvi2z3UGTM64JtRT7Qc1S {
}
.uimix-1Bvi392qBdMy93LGLcVymb {
box-sizing: border-box;
-webkit-font-smoothing: antialiased;
  position: absolute;
  left: 33px;
  top: 97px;
  width: 100px;
  height: 100px;
  opacity: 1;
  overflow: visible;
  background: #FFFFFF;
  border-style: solid;
  border-color: transparent;
  border-top-width: 0px;
  border-right-width: 0px;
  border-bottom-width: 0px;
  border-left-width: 0px;
  border-top-left-radius: 0px;
  border-top-right-radius: 0px;
  border-bottom-right-radius: 0px;
  border-bottom-left-radius: 0px;
  box-shadow: none;
}
.uimix-1Bvi39mct9UNYZ9eTCCRdV {
  position: absolute;
  left: 165px;
  top: 86px;
}
.uimix-1Bvi39mct9UNYZ9eTCCRdV-1Bvi2z3TpKXtTJLTRPTEd8 {
}
.uimix-1Bvi39mct9UNYZ9eTCCRdV-1Bvi2z3To8ddXo6R6ZftVD {
}
.uimix-1Bvi39mct9UNYZ9eTCCRdV-1BwspVKcGYvxVctYVzjPu6 {
  display: block;
}
.uimix-1Bvi39mct9UNYZ9eTCCRdV-1Bvi2z3UB1xUVg4Xz9WTpJ {
}
.uimix-1Bvi39mct9UNYZ9eTCCRdV-1Bvi2z3U3SiK6r7gf9vh7N {
}
.uimix-1Bvi39mct9UNYZ9eTCCRdV-1BxSDQc9suV4UZUVEQsy6r {
}
.uimix-1Bvi39mct9UNYZ9eTCCRdV-1BxSDQc9suV4UZUVEQsy6r-1Bvi2z3UGTM64JtRT7Qc1S {
}
.uimix-1BwWRgjHyEzJjsehRefVBE {
box-sizing: border-box;
-webkit-font-smoothing: antialiased;
  position: relative;
  margin-top: 0px;
  margin-right: 0px;
  margin-bottom: 0px;
  margin-left: 0px;
  width: max-content;
  height: max-content;
  opacity: 1;
  overflow: visible;
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  justify-content: flex-start;
  row-gap: 10px;
  column-gap: 10px;
  padding-left: 3px;
  padding-right: 3px;
  padding-top: 3px;
  padding-bottom: 3px;
  background: #f0f9ff;
  border-style: solid;
  border-color: transparent;
  border-top-width: 0px;
  border-right-width: 0px;
  border-bottom-width: 0px;
  border-left-width: 0px;
  border-top-left-radius: 8px;
  border-top-right-radius: 8px;
  border-bottom-right-radius: 8px;
  border-bottom-left-radius: 8px;
  box-shadow: none;
}
.uimix-1BwWRgjHr8HNgX8BRooGv7 {
box-sizing: border-box;
-webkit-font-smoothing: antialiased;
  position: relative;
  margin-top: 0px;
  margin-right: 0px;
  margin-bottom: 0px;
  margin-left: 0px;
  width: max-content;
  height: max-content;
  opacity: 1;
  overflow: visible;
  background: transparent;
  border-style: solid;
  border-color: transparent;
  border-top-width: 0px;
  border-right-width: 0px;
  border-bottom-width: 0px;
  border-left-width: 0px;
  border-top-left-radius: 0px;
  border-top-right-radius: 0px;
  border-bottom-right-radius: 0px;
  border-bottom-left-radius: 0px;
  box-shadow: none;
}
.uimix-1BwWRgjHmYZ8qLKDj3Tvdy {
box-sizing: border-box;
-webkit-font-smoothing: antialiased;
  position: relative;
  margin-top: 0px;
  margin-right: 0px;
  margin-bottom: 0px;
  margin-left: 0px;
  width: calc(100% - 0px);
  min-width: undefined;
  max-width: undefined;
  height: max-content;
  opacity: 1;
  overflow: visible;
  white-space: break-spaces;
  display: flex;
  flex-direction: column;
  color: #1f2937;
  font-family: Satoshi;
  font-size: 20px;
  font-weight: 700;
  line-height: normal;
  letter-spacing: 0px;
  text-align: start;
  justify-content: flex-start;
}
.uimix-1BwWRgjHuAe9UTxVhowwYu {
box-sizing: border-box;
-webkit-font-smoothing: antialiased;
  position: relative;
  margin-top: 0px;
  margin-right: 0px;
  margin-bottom: 0px;
  margin-left: 0px;
  width: calc(100% - 0px);
  min-width: undefined;
  max-width: undefined;
  height: max-content;
  opacity: 1;
  overflow: visible;
  white-space: break-spaces;
  display: flex;
  flex-direction: column;
  color: #4b5563;
  font-family: Inter;
  font-size: 16px;
  font-weight: 400;
  line-height: normal;
  letter-spacing: 0px;
  text-align: start;
  justify-content: flex-start;
}",
  "src/components.uimix.tsx": "import React from \\"react\\";
import * as header from \\"../../sandbox/src/stories/Header\\";
import * as button from \\"../../sandbox/src/stories/Button\\";
import image_YmY$tuteU03QY$k_2ARy1uacNmXDNmOX$FFAlj2qCIo from \\"../.uimix/images/YmY-tuteU03QY-k_2ARy1uacNmXDNmOX-FFAlj2qCIo.png\\";
import image_XMIUgNq9VxH1FiVdc4M95ZcfqaSP3aF07GtbcGHi9AY from \\"../.uimix/images/XMIUgNq9VxH1FiVdc4M95ZcfqaSP3aF07GtbcGHi9AY.png\\";
import image_DGLFxM8eMqBjHUQYEekprnB3xC$f5uvLxL7x2odVAEU from \\"../.uimix/images/DGLFxM8eMqBjHUQYEekprnB3xC-f5uvLxL7x2odVAEU.png\\";
import image_1yKf5NvwGW$XtLSBDlS3GqZfbEGnLshgQrjyQJImqaM from \\"../.uimix/images/1yKf5NvwGW-XtLSBDlS3GqZfbEGnLshgQrjyQJImqaM.png\\";
import \\"./components.uimix.css\\";

function applyOverrides(
  rootProps: any,
  element: React.ReactElement
): React.ReactElement {
  const { \\"data-refID\\": refID, ...props } = element.props;

  let additionalProps: any = rootProps;
  for (const id of refID ?? []) {
    additionalProps = additionalProps[id + \\"Props\\"] ?? {};
  }
  additionalProps = { ...additionalProps };
  for (const key of Object.keys(additionalProps)) {
    if (key.endsWith(\\"Props\\")) {
      delete additionalProps[key];
    }
  }

  const result = React.cloneElement(
    { ...element, props },
    {
      ...additionalProps,
      className: additionalProps.className
        ? props.className + \\" \\" + additionalProps.className
        : props.className,
      children:
        additionalProps.children ??
        React.Children.map(element.props.children, (child) => {
          if (React.isValidElement(child)) {
            return applyOverrides(rootProps, child);
          } else {
            return child;
          }
        }),
    }
  );
  return result;
}

export const Button = React.forwardRef((props, ref) => {
  return applyOverrides(
    { ...props, ref },
    <div className={\\"uimix-1Bvi2z3UPfi6W3L8nDt5VA\\"}>
      <div className={\\"uimix-1Bvi2z3UGTM64JtRT7Qc1S\\"} data-refID={[\\"text\\"]}>
        {\\"Button\\"}
      </div>
    </div>
  );
}) as React.FC<
  JSX.IntrinsicElements[\\"div\\"] & {
    textProps?: JSX.IntrinsicElements[\\"div\\"];
  }
>;

export const Stack = React.forwardRef((props, ref) => {
  return applyOverrides(
    { ...props, ref },
    <div className={\\"uimix-1Bvi2z3TfkSpYXW29daTN8\\"}>
      <div
        className={\\"uimix-1Bvi2z3TpKXtTJLTRPTEd8\\"}
        data-refID={[\\"item_0\\"]}
      ></div>
      <div
        className={\\"uimix-1Bvi2z3To8ddXo6R6ZftVD\\"}
        data-refID={[\\"item_1\\"]}
      ></div>
      <div
        className={\\"uimix-1BwspVKcGYvxVctYVzjPu6\\"}
        data-refID={[\\"hidden_Item\\"]}
      ></div>
      <div
        className={\\"uimix-1Bvi2z3UB1xUVg4Xz9WTpJ\\"}
        data-refID={[\\"item_2\\"]}
      ></div>
      <div className={\\"uimix-1Bvi2z3U3SiK6r7gf9vh7N\\"} data-refID={[\\"text\\"]}>
        {\\"Hello, world!\\"}
      </div>
      <div
        className={\\"uimix-1BxSDQc9suV4UZUVEQsy6r uimix-1Bvi2z3UPfi6W3L8nDt5VA\\"}
        data-refID={[\\"button\\"]}
      >
        <div
          className={
            \\"uimix-1BxSDQc9suV4UZUVEQsy6r-1Bvi2z3UGTM64JtRT7Qc1S uimix-1Bvi2z3UGTM64JtRT7Qc1S\\"
          }
          data-refID={[\\"button\\", \\"text\\"]}
        >
          {\\"Button\\"}
        </div>
      </div>
    </div>
  );
}) as React.FC<
  JSX.IntrinsicElements[\\"div\\"] & {
    item_0Props?: JSX.IntrinsicElements[\\"div\\"];
    item_1Props?: JSX.IntrinsicElements[\\"div\\"];
    hidden_ItemProps?: JSX.IntrinsicElements[\\"div\\"];
    item_2Props?: JSX.IntrinsicElements[\\"div\\"];
    textProps?: JSX.IntrinsicElements[\\"div\\"];
    buttonProps?: JSX.IntrinsicElements[\\"div\\"];
  }
>;

export const Page = React.forwardRef((props, ref) => {
  return applyOverrides(
    { ...props, ref },
    <div className={\\"uimix-1Bvi319YVxZELz4tEskN1S\\"}>
      <header.Header
        className={\\"uimix-1Bvi31VKu8gfy6XzJ8xrHA\\"}
        data-refID={[\\"header\\"]}
        {...{}}
      />
      <div className={\\"uimix-1Bvi365mtWZmXKDPeJTB4s\\"} data-refID={[\\"frame\\"]}>
        <button.Button
          className={\\"uimix-1Bvi32d9RMi3ES5kk2UvuM\\"}
          data-refID={[\\"button\\"]}
          {...{
            label: \\"Button1\\",
            primary: true,
            backgroundColor: \\"red\\",
            size: \\"large\\",
          }}
        />
        <div
          className={
            \\"uimix-1Bvi2z3UrU9VkawwAteXs8 uimix-1Bvi2z3UPfi6W3L8nDt5VA\\"
          }
          data-refID={[\\"instance\\"]}
        >
          <div
            className={
              \\"uimix-1Bvi2z3UrU9VkawwAteXs8-1Bvi2z3UGTM64JtRT7Qc1S uimix-1Bvi2z3UGTM64JtRT7Qc1S\\"
            }
            data-refID={[\\"instance\\", \\"text\\"]}
          >
            {\\"Override Name\\"}
          </div>
        </div>
        <img
          className={\\"uimix-1Bvi392qBdMy93LGLcVymb\\"}
          data-refID={[\\"image\\"]}
          src={image_YmY$tuteU03QY$k_2ARy1uacNmXDNmOX$FFAlj2qCIo}
          alt=\\"Image\\"
        />
        <div
          className={
            \\"uimix-1Bvi39mct9UNYZ9eTCCRdV uimix-1Bvi2z3TfkSpYXW29daTN8\\"
          }
          data-refID={[\\"stack\\"]}
        >
          <div
            className={
              \\"uimix-1Bvi39mct9UNYZ9eTCCRdV-1Bvi2z3TpKXtTJLTRPTEd8 uimix-1Bvi2z3TpKXtTJLTRPTEd8\\"
            }
            data-refID={[\\"stack\\", \\"item_0\\"]}
          ></div>
          <div
            className={
              \\"uimix-1Bvi39mct9UNYZ9eTCCRdV-1Bvi2z3To8ddXo6R6ZftVD uimix-1Bvi2z3To8ddXo6R6ZftVD\\"
            }
            data-refID={[\\"stack\\", \\"item_1\\"]}
          ></div>
          <div
            className={
              \\"uimix-1Bvi39mct9UNYZ9eTCCRdV-1BwspVKcGYvxVctYVzjPu6 uimix-1BwspVKcGYvxVctYVzjPu6\\"
            }
            data-refID={[\\"stack\\", \\"hidden_Item\\"]}
          ></div>
          <div
            className={
              \\"uimix-1Bvi39mct9UNYZ9eTCCRdV-1Bvi2z3UB1xUVg4Xz9WTpJ uimix-1Bvi2z3UB1xUVg4Xz9WTpJ\\"
            }
            data-refID={[\\"stack\\", \\"item_2\\"]}
          ></div>
          <div
            className={
              \\"uimix-1Bvi39mct9UNYZ9eTCCRdV-1Bvi2z3U3SiK6r7gf9vh7N uimix-1Bvi2z3U3SiK6r7gf9vh7N\\"
            }
            data-refID={[\\"stack\\", \\"text\\"]}
          >
            {\\"Hello, world!\\"}
          </div>
          <div
            className={
              \\"uimix-1Bvi39mct9UNYZ9eTCCRdV-1BxSDQc9suV4UZUVEQsy6r uimix-1BxSDQc9suV4UZUVEQsy6r\\"
            }
            data-refID={[\\"stack\\", \\"button\\"]}
          >
            <div
              className={
                \\"uimix-1Bvi39mct9UNYZ9eTCCRdV-1BxSDQc9suV4UZUVEQsy6r-1Bvi2z3UGTM64JtRT7Qc1S uimix-1BxSDQc9suV4UZUVEQsy6r-1Bvi2z3UGTM64JtRT7Qc1S uimix-1Bvi2z3UGTM64JtRT7Qc1S\\"
              }
              data-refID={[\\"stack\\", \\"button\\", \\"text\\"]}
            >
              {\\"Nested Override\\"}
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}) as React.FC<
  JSX.IntrinsicElements[\\"div\\"] & {
    headerProps?: Partial<React.ComponentProps<typeof header.Header>>;
    frameProps?: JSX.IntrinsicElements[\\"div\\"];
    buttonProps?: Partial<React.ComponentProps<typeof button.Button>>;
    instanceProps?: JSX.IntrinsicElements[\\"div\\"];
    imageProps?: JSX.IntrinsicElements[\\"div\\"];
    stackProps?: JSX.IntrinsicElements[\\"div\\"];
  }
>;

export const Card = React.forwardRef((props, ref) => {
  return applyOverrides(
    { ...props, ref },
    <div className={\\"uimix-1BwWRgjHrhVbbuTV7m36kM\\"}>
      <div className={\\"uimix-1BwWRgjHyEzJjsehRefVBE\\"} data-refID={[\\"icon\\"]}>
        <svg
          className=\\"uimix-1BwWRgjHr8HNgX8BRooGv7\\"
          data-refID={[\\"heroicons_code_bracket\\"]}
          fill=\\"none\\"
          height=\\"24\\"
          viewBox=\\"0 0 24 24\\"
          width=\\"24\\"
          xmlns=\\"http://www.w3.org/2000/svg\\"
        >
          <path
            d=\\"M17.25 6.75L22.5 12L17.25 17.25M6.75 17.25L1.5 12L6.75 6.75M14.25 3.75L9.75 20.25\\"
            stroke=\\"#0EA5E9\\"
            strokeLinecap=\\"round\\"
            strokeLinejoin=\\"round\\"
            strokeWidth=\\"1.5\\"
          />
        </svg>
      </div>
      <div className={\\"uimix-1BwWRgjHmYZ8qLKDj3Tvdy\\"} data-refID={[\\"title\\"]}>
        {\\"React Components Import/Export\\"}
      </div>
      <div
        className={\\"uimix-1BwWRgjHuAe9UTxVhowwYu\\"}
        data-refID={[\\"description\\"]}
      >
        {
          \\"Seamlessly work with React components, boosting your design-to-development workflow.\\\\n\\"
        }
      </div>
    </div>
  );
}) as React.FC<
  JSX.IntrinsicElements[\\"div\\"] & {
    iconProps?: JSX.IntrinsicElements[\\"div\\"];
    heroicons_code_bracketProps?: JSX.IntrinsicElements[\\"div\\"];
    titleProps?: JSX.IntrinsicElements[\\"div\\"];
    descriptionProps?: JSX.IntrinsicElements[\\"div\\"];
  }
>;
",
}
`;
